helios-testing/src/main/java/com/spotify/helios/testing/HeliosSoloDeployment.java

```diff
@@ -410,10 +436,78 @@ public void close() {
   }
 
   /**
+   * Undeploy jobs left over by {@link TemporaryJobs}. TemporaryJobs should clean these up,
+   * but sometimes a few are left behind for whatever reason.
+   */
+  private void undeployLeftoverJobs() {
```

Comment:
Hmm. This sounds like admitting defeat. "It doesn't work, so let's do it again!" Not sure I like it -- undeploying a second time because the first undeploy failed for reasons unknown.